### PR TITLE
Add update_by_script method

### PR DESCRIPTION
**Pull request by wangch079**
## Part of https://github.com/elastic/enterprise-search-team/issues/4158

This PR adds a `update_by_script` method to `ESIndex` class.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by timgrein** [comment]
Curious: For what do we need that?

**Comment by wangch079** [comment]
>Curious: For what do we need that?

Check [here](https://docs.google.com/document/d/1D3CeQQcDZkHsuGYOpfoVoXVFphlW27FpT_Rifw52wD0/edit#heading=h.db8oh0h6nx3p). It will be used for concurrency control.

File: connectors/tests/test_es_index.py [hunk 1]
```diff
@@ -182,6 +184,20 @@ async def test_update_with_concurrency_control(mock_responses):
...
+    script = {"source": ""}
+    index = ESIndex(index_name, config)
+    index.client = Mock()
+    index.client.update = AsyncMock()
+
+    await index.update_by_script(doc_id, script)
+    index.client.update.assert_awaited_once_with(
+        index=index_name, id=doc_id, script=script
+    )
+
```
Comment (navarone-feekery) [inline connectors/tests/test_es_index.py:199]: (Possibly ignorant) question: I see that this test is only checking that the `self.client.update` was called, so you've set the arg `script = {"source": ""}`.
I'm guessing that it isn't possible to test if an actual script would work, right?

Comment (wangch079) [reply connectors/tests/test_es_index.py:199]: >I'm guessing that it isn't possible to test if an actual script would work, right?

Could you elaborate? The test is just testing the correct params are passed to `update` function.

`if an actual script would work` sounds like it requires a real ES instance, which I think it's out of scope of unit test.

**Review by navarone-feekery — APPROVED** [review]
LGTM

**Pull request merged**